votor/src/event_handler/stats.rs

```diff
@@ -0,0 +1,502 @@
+            ),
+            (
+                "timeout_crashed_leader_elapsed_us",
+                self.timeout_crashed_leader.time_us as i64,
+                i64
+            ),
+            ("timeout_count", self.timeout.count as i64, i64),
+            ("timeout_elapsed_us", self.timeout.time_us as i64, i64),
+            ("safe_to_notar_count", self.safe_to_notar.count as i64, i64),
+            (
+                "safe_to_notar_elapsed_us",
+                self.safe_to_notar.time_us as i64,
+                i64
+            ),
+            ("safe_to_skip_count", self.safe_to_skip.count as i64, i64),
+            (
+                "safe_to_skip_elapsed_us",
+                self.safe_to_skip.time_us as i64,
+                i64
+            ),
+            (
+                "produce_window_count",
+                self.produce_window.count as i64,
+                i64
+            ),
+            (
+                "produce_window_elapsed_us",
+                self.produce_window.time_us as i64,
+                i64
+            ),
+            ("finalized_count", self.finalized.count as i64, i64),
+            ("finalized_elapsed_us", self.finalized.time_us as i64, i64),
+            ("standstill_count", self.standstill.count as i64, i64),
+            ("standstill_elapsed_us", self.standstill.time_us as i64, i64),
+            ("set_identity_count", self.set_identity.count as i64, i64),
+            (
+                "set_identity_elapsed_us",
+                self.set_identity.time_us as i64,
+                i64
+            ),
+        );
+    }
+}
+
+#[derive(Debug, Default)]
+struct SentVoteStats {
+    finalize: usize,
+    notarize: usize,
+    notarize_fallback: usize,
+    skip: usize,
+    skip_fallback: usize,
+}
+
+impl SentVoteStats {
+    fn incr_vote(&mut self, vote_type: VoteType) {
+        match vote_type {
+            VoteType::Finalize => self.finalize = self.finalize.saturating_add(1),
+            VoteType::Notarize => self.notarize = self.notarize.saturating_add(1),
+            VoteType::NotarizeFallback => {
+                self.notarize_fallback = self.notarize_fallback.saturating_add(1)
+            }
+            VoteType::Skip => self.skip = self.skip.saturating_add(1),
+            VoteType::SkipFallback => self.skip_fallback = self.skip_fallback.saturating_add(1),
+        }
+    }
+
+    fn report(&self) {
+        datapoint_info!(
+            "event_handler_sent_vote_count",
+            ("finalize", self.finalize as i64, i64),
+            ("notarize", self.notarize as i64, i64),
+            ("notarize_fallback", self.notarize_fallback as i64, i64),
+            ("skip", self.skip as i64, i64),
+            ("skip_fallback", self.skip_fallback as i64, i64),
+        );
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct EventHandlerStats {
+    // Number of events that were ignored. This includes events that were
+    // received but not processed due to various reasons (e.g., outdated,
```

Comment:
nit: Myself and future people appreciate the comments. One nit - you used `///` above but `//` here; I guess triple might be more proper ? I'm not really sure nor do I have a strong preference, but I do like consistency